### PR TITLE
test(YouTube): Add integration test

### LINKED ISSUE #130: youtube-integration-test
Hooray 🎉 thanks for your interest in [MDX Embed](https://mdx-embed.com) 

If you'd like to contribute by writing this test please have a read of: 

- The [Umbrella Issue](https://github.com/PaulieScanlon/mdx-embed/issues/71) 
- The [Acceptance Criteria](https://github.com/PaulieScanlon/mdx-embed/issues/71#acceptance-criteria) for **Integration Tests**

### PR DESCRIPTION
**Pull request by JeffersonBledsoe**
Adds tests for the following:
- [x] The YouTube iframe displaying
- [ ] A valid id prop being passed doesn't produce a video error

Closes #130

### CODE REVIEW

**Comment by JeffersonBledsoe** [comment]
Trying to write a test to click the 'play' button on the YouTube video and check it doesn't display the error message, it doesn't seem like it's able to find and interact with elements inside the iframe. The following test fails:

```js
getIframeBody().find('.ytp-large-play-button').click();
getIframeBody().find('.ytp-error-icon-container').should('be.undefined');
```

```
Timed out retrying: Expected to find element: .ytp-large-play-button, but never found it. Queried from element: <body>
```

Even though this test passes:

```js
getIframeBody().find('.ytp-large-play-button').should('not.be.undefined');
```

**Comment by PaulieScanlon** [comment]
Hey @JeffersonBledsoe are you able to push the `.spec` file it's difficult to determine where the issue might be without seeing the whole thing... is that cool?

I wonder if the YouTube play button is nested within another iframe which can be tricky to manage. Storybook itself runs in an iframe which is what that util function is for. If the Component is then loaded into another iframe we end up in a kind of inception iframe whirlpool of hell! 

For what it's worth i wouldn't worry about testing the play button, that's YouTube's job to make sure their widgets work. 

All we're really concerned about is "do they load" ... i talked a bit more about this here if it's at all helpful: https://youtu.be/TzWm_tullLo?t=555

and thanks for taking the time to add the comments

File: packages/mdx-embed/src/components/youtube/youtube.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export const YouTube: FunctionComponent<IYouTubeProps> = ({
         }}
       >
         <iframe
+          data-testid={`youTube-${youTubeId}`}
```
Comment (PaulieScanlon) [inline packages/mdx-embed/src/components/youtube/youtube.tsx:44]: The `data-testid` is only required for unit testing.

**Review by PaulieScanlon — CHANGES_REQUESTED** [review]

File: packages/mdx-embed/src/components/youtube/youtube.tsx (hunk 1 above)
Comment (JeffersonBledsoe) [reply packages/mdx-embed/src/components/youtube/youtube.tsx:44]: My bad, I pushed the wrong bit of code! 🤦‍♂️

Comment (PaulieScanlon) [reply packages/mdx-embed/src/components/youtube/youtube.tsx:44]: no worries man! 😎

**Comment by JeffersonBledsoe** [comment]
@PaulieScanlon Thanks for the link, the reason behind just testing if the iframe loads is much clearer now! I'm happy to remove the play button test, the last thing we want is to deal with more iframes! 😄

**Comment by PaulieScanlon** [comment]
@JeffersonBledsoe cool man, glad that cleared things up. We may well need to elaborate on these integration tests at a later date but our goal for a stable release for v1 is just to have every component covered in some way by a unit test and a basic integration test. Plus for Hacktoberfest i didn't want to complicate matters. There's now 6 active contributors and 40 issues... and i want to go to the pub at some point 😎

Great job so far and if you wanna take on more please do!

**Review by PaulieScanlon — APPROVED** [review]

**Pull request merged**